### ISSUE TITLE
/www/build/* in source control?

### ISSUE THREAD

**Issue by kylecordes** [ionic-team/stencil-component-starter#10]
Currently there is a build output, the contents of the build directory, committed to the repo. This is unusual, and often represents a missed .gitignore configuration. Are these files present in the source control intentionally? If so, may be explained in the readme? If not, fix?

(I'm happy to do a PR to fix or add an explanation, if pointed as to which one is right here)

**Comment by siedentm** [comment]
as it stands, www is getting generated. it could be left out completely i believe.

**Comment by kylecordes** [comment]
Is this something you want a PR on? Or something a core dev prefers to do?

**Comment by siedentm** [comment]
actually, i just played around with it. www is required (the index.html is getting built). 

when using this config, stencil stops working, once the www folder gets deleted:
```
exports.config = {
  namespace: 'myname',
  generateDistribution: true,
  generateWWW: false,
  hashFileNames: false,
  bundles: [
    { components: ['my-name'] }
  ]
};
```
in addition to that, index.html is always going to be generated. this seems weird and shouldn't be a problem of the component starter itself but stencil.

**Comment by siedentm** [comment]
well, the main question is: is this behavior intended? what use is there for www when let's say just importing @stencil/core without the dev server in order to incorporate it into a larger project or plainly just for generating collections and/or component output files? the www directory seems to be only useful during the dev stage. I believe it's unwanted behavior to have the www directory for the component starter at distribution time while it does indeed makes sense for developing complete apps.
I posted a bug report and mentioned the www folder as well in the stencil project [(here)](https://github.com/ionic-team/stencil/issues/188)

**Comment by peterennis** [comment]
Look here:
https://github.com/ionic-team/stencil-starter/commit/c497bf4ea96057d16750713453551c2ba6eebb0e

![capture1102](https://user-images.githubusercontent.com/140737/30935874-e2f8fe1c-a386-11e7-9619-fb6b947d7a64.PNG)

@adamdbradley 
I still haven't totally wrapped my brain around the difference with the stencil-starter (app) and stencil-component-starter and wonder if they should be kept in synch. Ref: https://github.com/ionic-team/stencil-starter/issues/12

**Comment by kylecordes** [comment]
@siedentm  As far as I know, the only reason you would want something like a www directory in a kit made for building components for distribution, is to conveniently demo, develop, and test those components in isolation. And that is a very good reason. But it doesn't seem like a reason why build output should be committed to a repo. It seems like maybe some adjustment is needed to make the build process here, when setting up a local www directory for develop/demo, "just work" without requiring some of these files be preserved?

**Comment by siedentm** [comment]
@kylecordes i totally agree. as it looks, we're getting a clean stencil-compiler version that does not depend on www since the www-requirement got classified as a bug in stencil (not stencil-component-starter). Let them fix it, then we can generate www for --dev and ignore it for distribution builds.

**Comment by kensodemann** [comment]
> Perhaps there should be a www/.gitignore, which ensures that the directory exists but avoids checking in all the stuff inside it?

Done. I did this as part of the clean-up I did when I added the unit testing stuff (actually, I first completely removed www/ from version control then discovered that broke things, so then added the `www/.gitignore`).

FWIW, as a developer, I would want the www/ directory here for use during dev, and I find it strange that it must pre-exist (obviously, since my first reaction was to remove it and break things).
